cub/cub/device/dispatch/kernels/transform.cuh

```diff
@@ -125,6 +128,125 @@ _CCCL_DEVICE void transform_kernel_impl(
   }
 }
 
+template <typename T, int LoadLength>
+using vector_type = ::cuda::std::_If<
+  sizeof(T) * LoadLength == 1,
+  int8_t,
+  ::cuda::std::_If<
+    sizeof(T) * LoadLength == 2,
+    int16_t,
+    ::cuda::std::_If<
+      sizeof(T) * LoadLength == 4,
+      int32_t,
+      ::cuda::std::_If<
+        sizeof(T) * LoadLength == 8,
+        int64_t,
+        ::cuda::std::_If<sizeof(T) * LoadLength == 16,
+                         int4,
+                         ::cuda::std::_If<sizeof(T) * LoadLength == 32, longlong4, ::cuda::std::array<T, LoadLength>>>>>>>;
+
+// This kernel guarantees that objects passed as arguments to the user-provided transformation function f reside in
+// global memory. No intermediate copies are taken. If the parameter type of f is a reference, taking the address of
+// the parameter yields a global memory address.
+template <typename VectorizedPolicy,
+          typename Offset,
+          typename F,
+          typename RandomAccessIteratorOut,
+          typename... RandomAccessIteratorIn>
+_CCCL_DEVICE void transform_kernel_impl(
+  ::cuda::std::integral_constant<Algorithm, Algorithm::vectorized>,
+  Offset num_items,
+  int /*num_elem_per_thread*/,
+  F f,
+  RandomAccessIteratorOut out,
+  RandomAccessIteratorIn... ins)
+{
+  static_assert((::cuda::std::contiguous_iterator<RandomAccessIteratorIn> && ...));
+  static_assert(::cuda::std::contiguous_iterator<RandomAccessIteratorOut>);
+
+  constexpr int block_dim        = VectorizedPolicy::block_threads;
+  constexpr int items_per_thread = VectorizedPolicy::items_per_thread;
+  constexpr int tile_stride      = block_dim * items_per_thread;
+  const Offset offset            = static_cast<Offset>(blockIdx.x) * tile_stride;
+  const int tile_size            = static_cast<int>((::cuda::std::min)(num_items - offset, Offset{tile_stride}));
+
+  // move index and iterator domain to the block/thread index, to reduce arithmetic in the loops below
+  {
+    (..., (ins += offset));
+    out += offset;
+  }
+
+  // Disabling prefetching is better on the A100
+  // (..., prefetch_tile<block_dim>(ins, tile_size));
+
+  // FIXME(bgruber): implement handling of unalinged data
+
+  if (tile_size == tile_stride)
```

Comment:
Nitpick: the else branch is much shorter and could short circuit improving readability of the remaining code

Reply:
Obsolete, handled at the beginning now by delegating to the prefetch kernel.